lib/command/src/Obelisk/Command/Run.hs

```diff
@@ -88,21 +86,45 @@ run = do
       , "Frontend.frontend"
       ]
 
+-- | Nix syntax requires relative paths to be prefixed by @./@ or
+-- @../@. This will make a 'FilePath' that can be embedded in a Nix
+-- expression.
+toNixPath :: FilePath -> FilePath
+toNixPath root | "/" `isInfixOf` root = root
+               | otherwise = "./" <> root
+
 runRepl :: MonadObelisk m => m ()
 runRepl = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> do
     runGhciRepl dotGhciPath
 
 runWatch :: MonadObelisk m => m ()
 runWatch = do
-  pkgs <- getLocalPkgs
+  pkgs <- withProjectRoot "." getLocalPkgs
   withGhciScript pkgs $ \dotGhciPath -> runGhcid dotGhciPath Nothing
 
--- | Relative paths to local packages of an obelisk project
--- TODO a way to query this
-getLocalPkgs :: Applicative f => f [FilePath]
-getLocalPkgs = pure ["backend", "common", "frontend"]
+-- | Relative paths to local packages of an obelisk project.
+--
+-- These are a combination of the obelisk predefined local packages,
+-- and any packages that the user has set with the @packages@ argument
+-- to the Nix @project@ function.
+getLocalPkgs :: (MonadObelisk m, MonadIO m) => FilePath -> m [FilePath]
+getLocalPkgs root = do
+  root' <- liftIO $ makeAbsolute root
+  -- package paths from nix project will be absolute paths
+  projectPackages <- readProcessJSONAndLogStderr Debug $
+    proc "nix"
+      [ "eval"
+      , "(let proj = import " <> root' <> " {}; in (map toString (proj.obelisk.reflex-platform.nixpkgs.lib.attrValues (proj.passthru.packages or {}))))"
```

Comment:
I guess here (and elsewhere) we are very sloppy with escaping... we should at least stub it out. There is an issue that's *almost* right for this (#476) but that's for shell escaping, not for nix expression escaping...

Reply:
I *assume* this gives non-nix paths for some projects? If not then I'm not sure how it actually helps...

Reply:
I have changed it to import `root` (the relative path) instead of `root'` (the absolute path). So if there are spaces or nix syntax in the project directory it shouldn't matter.

> I assume this gives non-nix paths for some projects? If not then I'm not sure how it actually helps...

Could you please clarify what this means? I don't understand.

Reply:
I mean that `projectPackages` is a list of paths, but none of them are in the nix store?